### PR TITLE
FEATURE: Add additional factory methods `createWithAllowedNodeTypeNames` and `createWithDisallowedNodeTypeNames` to the nodeTypeConstraints.

### PR DESCRIPTION
**Pull request by mficzel**
This makes defining constraints easier forward as one does not have to specify empty constraints unused conditions.

In addition the arguments names of `nodeTypeConstraints::create` were shortened and the use via named arguments is recommended. 

**Checklist**

- [x] Code follows the PSR-2 coding style
- [x] Tests have been created, run and adjusted as needed
- [x] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/ParentsOperation.php [hunk 1]
```diff
@@ -71,8 +71,7 @@ public function evaluate(FlowQuery $flowQuery, array $arguments)
     {
         $parents = Nodes::createEmpty();
         $findAncestorNodesFilter = FindAncestorNodesFilter::create(
-            NodeTypeConstraints::create(
-                NodeTypeNames::createEmpty(),
+            NodeTypeConstraints::createWithDisallowedNodeTypeNames(
                 NodeTypeNames::fromArray(
                     [NodeTypeName::fromString('Neos.ContentRepository:Root')]
                 )
```
Comment (bwaidelich) [inline Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/ParentsOperation.php]:
```suggestion
                NodeTypeNames::fromStringArray(['Neos.ContentRepository:Root')])
```

**Review by bwaidelich — DISMISSED** [review]
+1 by reading with a tiny suggestion you are free to ignore :)

**Review by mhsdesign — APPROVED** [review]
That looks great ;)

I still like to improve the docs of `create` to inform that it should be used with named arguments and also shorten its params so they are suitable for named arguments.

**Review by mhsdesign — APPROVED** [review]
Thanks i like this ;)

I like to rerequest bastians review after this change though ...

**Review by bwaidelich — APPROVED** [review]
Love it, thanks!

**Pull request merged**